include/podio/EventStore.h

```diff
@@ -112,16 +112,16 @@ class EventStore : public ICollectionProvider, public IMetaDataProvider {
   bool doGet(const std::string& name, CollectionBase*& collection, bool setReferences = true) const;
   /// check if a collection of given name already exists
   bool collectionRegistered(const std::string& name) const;
-  void setCollectionIDTable(CollectionIDTable* table) {
-    m_table.reset(table);
+  void setCollectionIDTable(std::shared_ptr<CollectionIDTable> table) {
+    m_table = std::move(table);
   }
 
   // members
   mutable std::set<int> m_retrievedIDs{};
   mutable CollContainer m_collections{};
   mutable std::vector<CollectionBase*> m_cachedCollections{};
   IReader* m_reader{nullptr};
-  std::unique_ptr<CollectionIDTable> m_table;
+  std::shared_ptr<CollectionIDTable> m_table;
```

Comment:
Essentially this is needed for the read tests, which now also try to jump to some events and make sure that works. In the course of this the reader is only created once, but the `EventStore` is created multiple times. If the `EventStore` owns this uniquely, the first `EventStore` to go out of scope also destroys the table that is held by the reader.

The current design is overall a bit backwards I think. See also #314

Having (unique) ownership with the readers doesn't really work either, because `EventStore`s can be created without a reader, and in that case they have to own the table, because otherwise ownership would have to go to the writers, which again breaks, because there can be multiple writers to each `EventStore`.

This should all be solved much more cleanly with the Frame...